### ISSUE TITLE
<Javascript Course>: <Battleship project needs to be more concise>

### ISSUE THREAD

**Issue by SilviuHg** [TheOdinProject/curriculum#27155]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

In the Battleship project, upon reaching step 3, I realised the explanation was a bit vague:
3. Create Player.
1. Players can take turns playing the game by attacking the enemy Gameboard.
2. The game is played against the computer, so make the ‘computer’ capable of making random plays. The AI does not have to be smart, but it should know whether or not a given move is legal (i.e. it shouldn’t shoot the same coordinate twice).
After reading the suggestion, I was not sure of what I need to accomplish. I called for help on the discord server, asking what the purpose of the player factory was, and one person told me that there are 2 players, and each player has it's own board, where they can place ships and receive attacks. So besides the gameboard function, i needed another player function that would act as my AI gameboard (place the AI ship coords, receive it's attacks etc). We both agreed that TOP is kinda vague regarding the player function. I didn't know if it was vague on purpose or not, so I chose to write this issue. Thank you!

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-javascript-battleship

### (Optional) Discord Name

special4975

**Comment by JoshDevHub** [comment]
Thank you for making this issue @SilviuHg 

@TheOdinProject/javascript Any thoughts on this issue?

**Comment by 01zulfi** [comment]
@SilviuHg Thanks for making this issue.

I'm not sure what's vague here. It's clear that the battleship game is played between two players (we even link the Wikipedia article at the start of the project) who have their own boards and ship placements.

**Comment by SilviuHg** [comment]
@01zulfi The structure of the project is, in my opinion, the most important. So, you take your time and think how the project should look/what every module does. In the battleship project, I felt like the structure was a little confusing for me (maybe I'm just too unexperienced so that could be the reason). TOP suggests that the gameboard object should have the following methods: place ships, receive attacks, keep track of attacks, report if ships are sunk. Going into the player object, no such suggestion is given anymore. It just states how the game is played. 

After some discussions and a little thought, I realised the player object should help in initialising a player that has it's own gameboard and it's own methods (like intentional or random attacks). I felt like little guidance was given for the player method, and for me at least was a bit confusing. At the time of my discord discussions there was one other person that too seemed confused and told me to keep asking questions because it was helping him too. 

As I stated in my original post, I was not sure if TOP was intentionally giving less 'guidance' on that specific step, so I thought a question would never hurt :D. Thanks for your reply.
